nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -81,14 +73,53 @@ export async function conditionalMakeBlock(proposer) {
     or we're no-longer the proposer (boo).
    */
   if (proposer.isMe) {
-    const unprocessed = await numberOfUnprocessedTransactions();
-    let numberOfProposableL2Blocks = Math.floor(unprocessed / TRANSACTIONS_PER_BLOCK);
-    // if we want to make a block right now but there aren't enough transactions, this logic
-    // tells us to go anyway
-    if (makeNow && unprocessed > 0 && numberOfProposableL2Blocks === 0)
-      numberOfProposableL2Blocks = 1;
-
-    if (numberOfProposableL2Blocks >= 1) {
+    // Get all the mempool transactions sorted by fee
+    const mempoolTransactions = await getSortedMempoolTransactions();
```

Comment:
can we rename this to show this is sorted by fee, e.g. `getSortedByFeeMempoolTransactions()`